crates/tools/gnu/src/main.rs

```diff
@@ -1,28 +1,38 @@
-use std::collections::BTreeMap;
+use std::collections::{BTreeMap, BTreeSet};
 use std::io::prelude::*;
 
 fn main() {
-    let mut cmd = std::process::Command::new("where");
-    cmd.arg("dlltool.exe");
-
-    let output = cmd.output().unwrap();
-
-    if !output.status.success() {
-        println!("dlltool.exe not found");
-        return;
+    const ALL_PLATFORMS: [&str; 4] = ["x86_64_gnu", "i686_gnu", "x86_64_gnullvm", "aarch64_gnullvm"];
+    let mut platforms = BTreeSet::new();
+    for platform in std::env::args().skip(1) {
+        if ALL_PLATFORMS.contains(&&*platform) {
+            platforms.insert(platform);
+        } else if platform == "all" {
+            platforms.extend(ALL_PLATFORMS.iter().map(|s| s.to_string()));
+        } else {
+            eprintln!("Unknown platform: {}", platform);
+            return;
+        }
     }
-
-    let output = unsafe { core::str::from_utf8_unchecked(&output.stdout) };
-
-    let platform = if output.contains("mingw64") {
-        "x86_64_gnu"
-    } else if output.contains("mingw32") {
-        "i686_gnu"
-    } else {
-        println!("mingw not found");
+    if platforms.is_empty() {
+        eprintln!("Please specify at least one platform or use 'all' argument");
         return;
     };
 
+    for platform in platforms {
+        let tools = if platform.ends_with("_gnu") { &["dlltool", "ar", "objcopy"][..] } else { &["llvm-dlltool", "llvm-ar"][..] };
+        for tool in tools {
+            if which::which(tool).is_err() {
+                eprintln!("Could not find {}. Is it in your $PATH?", tool);
+                return;
+            }
```

Comment:
We can remove the `which` dependency by just trying to invoke the tool. Alternatively, we could do nothing at all and just fail too, given this is mostly maintainer tooling. I think either approach is preferable over the dependency hit.

```suggestion
            if Command::new(tool).stdout(Stdio::null()).spawn().is_err() {
                eprintln!("Could not find {}. Is it in your $PATH?", tool);
                return;
            }
```